roles/common/tasks/monitoring.yml

```diff
@@ -50,18 +50,57 @@
   notify: restart sensu-client
   when: monitoring.client_key is defined
 
+- set_fact:
+    ruby_path: "/opt/sensu/embedded/bin/"
+    default_path: /etc/default/sensu
+  when: openstack_install_method != 'distro'
+  
+- set_fact:
+    ruby_path: ""
+    default_path: /etc/sysconfig/sensu
+  when: openstack_install_method == 'distro'
+
 - name: install dependancy gems
-  shell: /opt/sensu/embedded/bin/gem install {{ item }} --no-user-install
+  shell: "{{ ruby_path }}gem install {{ item }} --no-user-install"
+  register: result
+  until: result|succeeded
+  retries: 5
   with_items: "{{ monitoring.checks.gems }}"
 
+# block for rhel distro  
+- block:
+  - name: install dependancy gems for rhosp distro
+    shell: "{{ ruby_path }}gem install {{ item.name }} --no-user-install {{ '-v ' if item.version is defined else ' ' }} {{ item.version|default(' ') }}"
+    register: result
+    until: result|succeeded
+    retries: 5
+    with_items: "{{ monitoring.checks.gems_rhel_distro }}"
+
+  - name: create path for sensu embedded ruby
+    file: dest=/opt/sensu/embedded/bin state=directory owner=sensu mode=0750
+
+  - name: add symbolic link for embedded ruby path (to compatible with some sensu-plugins's hardcode embeded ruby path)
```

Comment:
now that we're setting PATH in /etc/default/sensu on Ubuntu can we instead remove the hardcoded /opt/sensu/embedded/bin/ruby from all the shebangs in the checks  ?
Looking them over and it seems feasible.